### ISSUE TITLE
Running concurrent experiments on same endpoint/view

### ISSUE THREAD

**Issue by karmakaze** [splitrb/split#405]
As I understand how the Split gem works, by default a user can only participate in one experiment at a time. Let's say we have a page (or an API endpoint) with two experiments running on it:

my_controller.rb
```ruby
  def show
    alt = ab_test(:test1, 'control', 'a')
    response.headers['X-split-test1'] = alt

    alt = ab_test(:test2, 'control', 'x')
    response.headers['X-split-test2'] = alt
```

With this setup, `test2` always reports the control group `control`, presumably until `test1` has a winner.

Enabling `allow_multiple_experiments` will have the undesired effect of allowing a given user to belong to non-control groups in multiple experiments. Is this the case, and if so has any thought been given to this?

**Comment by andrew** [comment]
Yeah this is the case I've not run into that as I don't run many experiments at the same time myself but that could definitely be an issue, any ideas on how to fix/improve it?

**Comment by andreibondarev** [comment]
@karmakaze Is the requirement to enroll a user in test1 XOR test2 experiment?

**Comment by karmakaze** [comment]
Yes, the existing default behaviour of only allowing a user to be in a test group is desired if we consider being in the 'control' test group to be non-exclusive.

**Comment by andreibondarev** [comment]
@karmakaze Would you be able to achieve this by combining the 2 tests? `alt = ab_test(:test1_and_2, 'control', 'a', 'x')`

**Comment by karmakaze** [comment]
The point of having a testing framework is to allow easy running of tests. Test1 and test2 could have very different but overlapping lifetimes. I'm going to see if I can put a PR together that has special handling for the 'control' alternative value, and make that an option between single tests and multiple test options.

**Comment by andrew** [comment]
@karmakaze I've just released 1.7.0 for you: https://github.com/splitrb/split/releases/tag/v1.7.0
